staging_service/import_specifications/file_parser.py

```diff
@@ -166,14 +166,58 @@ def __post_init__(self):
 
 def parse_import_specifications(
     paths: tuple[Path],
-    file_type_resolver: Callable[[Path], FileTypeResolution]
+    file_type_resolver: Callable[[Path], FileTypeResolution],
+    log_error: Callable[[Exception], None]
 ) -> ParseResults:
     """
     Parse a set of import specification files and return the results.
 
     paths - the file paths to open.
     file_type_resolver - a callable that when given a file path, returns the type of the file or
         a parser for the file.
+    log_error - callable for logging an exception.
     """
+    if not paths:
+        return ParseResults(errors=tuple([Error(ErrorType.NO_FILES_PROVIDED)]))
+    # TODO check for None, whitespace only paths, '.', '..' & Path.resolve()
+    try:
+        return _parse(paths, file_type_resolver)
+    except Exception as e:
+        log_error(e)
+        errors = tuple([Error(ErrorType.OTHER, str(e))])
+        return ParseResults(errors=errors)
+
+def _parse(
+    paths: tuple[Path],
+    file_type_resolver: Callable[[Path], FileTypeResolution],
+) -> ParseResults:
     results = {}
     errors = []
+    for p in paths:
+        file_type = file_type_resolver(p)
+        if file_type.unsupported_type:
```

Comment:
I wonder if its better to check this attribute or have the file_type_resolver be in charge of throwing the exception.

Reply:
I did it here for a couple reasons:

* The user of the method needs to write slightly less code
* At the end of the method, the user gets all the errors in one list in a common format